src/VisualStudio/Core/Def/Implementation/DesignerAttribute/DesignerAttributeIncrementalAnalyzer.cs

```diff
@@ -186,13 +209,17 @@ private bool CheckVersions(Document document, VersionStamp textVersion, VersionS
                 return;
             }
 
+            await RegisterDesignerAttributeAsync(document, designerAttributeArgumentOpt.Value, cancellationToken).ConfigureAwait(false);
+
+            // persist the information locally *after* we try to register with the project system.
+            // Registering can fail (i.e. if the project is unloaded), and we don't want to persist
+            // a value that makes us think we've communicated this properly to the project system.
             var data = new Data(textVersion, semanticVersion, designerAttributeArgumentOpt.Value);
             await _state.PersistAsync(document, data, cancellationToken).ConfigureAwait(false);
-
-            await RegisterDesignerAttributeAsync(document, designerAttributeArgumentOpt.Value, cancellationToken).ConfigureAwait(false);
```

Comment:
note: this new code matches the name of the method now (`RegisterDesignerAttributeAndSaveStateAsync`).  Previously, it was saving state, and then registering the attribute.  Now we actually register the attribute *then* save the state.